src/lib.rs

```diff
@@ -86,8 +90,280 @@ mod sealed {
+    fn register(&self, child: &std::process::Child) -> io::Result<()> {
+        self.pipe.register(child)
+    }
+
+    /// Push a new "zombie" process into the reactor.
+    fn push_zombie(&'static self, mut child: std::process::Child) {
+        // Spawn the "async-process" thread if it hasn't been spawned yet.
+        self.spawn_thread();
+
+        let mut zombies = self.zombies.lock().unwrap();
+
+        // If the child process has already exited, then we don't need to push it into the list of zombies.
+        if let Ok(None) = child.try_wait() {
+            zombies.push(child);
+        }
+    }
+
+    /// Poll the reactor for "zombie" processes.
+    async fn poll(&self, _guard: MutexGuard<'_, ()>) -> ! {
+        loop {
+            // Wait for the next SIGCHLD signal.
+            self.pipe.wait().await;
+
+            // Notify all listeners waiting on the SIGCHLD event.
+            self.sigchld.notify(std::usize::MAX);
+
+            // Reap zombie processes.
+            let mut start = 0;
+            while let Some(new_start) = self.process_zombies(start) {
+                start = new_start;
+                future::yield_now().await;
+            }
+        }
+    }
+
+    /// Wait on zombie processes.
+    ///
+    /// Returns `Some(index to start at)` if we handled too many in one go.
+    fn process_zombies(&self, start: usize) -> Option<usize> {
+        let mut i = start;
+        let mut zombies = self.zombies.lock().unwrap();
+
+        // Only process a limited number of zombies at a time.
+        for _ in 0..100 {
+            // Get the zombie process.
+            let zombie = zombies.get_mut(i)?;
+
+            // Try to wait on it. If it's done, remove it from the list.
+            if let Ok(None) = zombie.try_wait() {
+                i += 1;
+            } else {
+                zombies.swap_remove(i);
+            }
+        }
+
+        // Yield to avoid starving other tasks.
+        Some(i + 1)
+    }
+
+    /// Spawn a backup thread to poll the reactor.
+    fn spawn_thread(&'static self) {
+        // Check to see if no one else is polling the reactor.
+        if let Some(guard) = self.polling_guard.try_lock() {
+            // If no one else is polling the reactor, then spawn a backup thread to poll it.
+            thread::Builder::new()
+                .name("async-process".to_string())
+                .spawn(move || {
+                    #[cfg(not(windows))]
+                    async_io::block_on(self.poll(guard));
+
+                    #[cfg(windows)]
+                    future::block_on(self.poll(guard));
+                })
+                .expect("cannot spawn async-process thread");
+        }
+    }
+}
+
+cfg_if::cfg_if! {
+    if #[cfg(windows)] {
+        /// A callback that simulates a SIGCHLD signal on Windows.
+        struct Pipe {
+            /// The event that is signalled to simulate a SIGCHLD signal.
+            event: Event,
```

Comment:
I'm not sure if an `Event` is semantically equivalent to the old `wait_sigchld` function on Windows.